video_sampler/sampler.py

```diff
@@ -99,9 +99,25 @@ def sample(self, video_path: str, subs: str = None) -> Iterable[list[FrameObject
             if self.cfg.keyframes_only:
                 stream.codec_context.skip_frame = "NONKEY"
             prev_time = -10
-            for frame_indx, frame in enumerate(container.decode(stream)):
+            
+            # Confirm end_frame is greater than start_frame
+            if self.cfg.end_frame is not None:
+                if self.cfg.end_frame <= self.cfg.start_frame:
+                    raise ValueError(f"Invalid end_frame: {self.cfg.end_frame}. Must be greater than start_frame")
+            
+            # Seek the starting point of the processing interval
+            container.seek(self.cfg.start_frame, stream=stream)
```

Comment:
I think we may need to seek to a proper time, rather than the frame number?
PyAV calls for `time_base` units which we can get from the stream: `time_base = stream.time_base`.
https://pyav.org/docs/develop/api/container.html#av.container.InputContainer.seek